docs/src/basics/nomenclature.md

```diff
@@ -0,0 +1,53 @@
+
+_Follower_:  In a clustered environment, a broker which is not a leader is a follower of a given partition.  A follower assumes leader responsibilities when the leader goes down.
+
+_Record_: A Zeebe record is a collection of workflow variables and other state transitions of an executing workflow instance at a given point in time.  During execution of a workflow instance numerous records will be generated to capture the various state transitions from start to workflow instance completion.
+
+_Event_:  Within the context of Zeebe, an event represent a state transition associated with an executing workflow instance.  Events include variable changes, changes to transition elements in the workflow model, etc.   An event will be represented by a timestamp, the variable name and variable value.  These events are stored in an append-only log.
+
+_Log_:  An ordered sequence of events written in persistent storage. This log file is only appended-to and is stored on disk.  
+
+_Segment_:  Data within a log file is segments.  Segments are defined by time period.???
+
+_Partition_:  A partition represents a logical grouping of data in as Zeebe broker.  This data includes workflow instance variables and events generated by Zeebe.  Partitions are determined by ???? 
+
+_Replication_:  In a clustered Zeebe installation configured with more than one broker, data in a partition is copied to the other brokers.  This act of copying data in partitions from one broker to another is also referred to as replication.
+   
+_Replication Factor_:  This is the number of times data in a partition will be shared and this depends on the number of brokers in a cluster.  The concept of sharing here implies that other brokers should have the exact same copy of data in all partitions.  Data replication allows the system to be resilient to brokers going down. 
+
+_Snapshot_:  The state of all active workflows instances, (these are also known as inflight workflow instances) are stored as records in an in-memory database called  RocksDb.  All events generated as a result of execution of a workflow instance in Zeebe are stored in an append-only log file.  A **snapshot** represents a copy of all data within a partition (including event logs and workflow instance variables in RocksDb) at any given point in time.  Snapshots are binary images stored on disk and can be used to restore execution state of a workflow.   The size of a snapshot is affected by the size of the data.  Size of the data depends on several factors including complexity of the model or business process, the size and quantity of variables in each workflow instance as well as the total number of executing workflow instances in a broker.
+
+_Exporter_:  An exporter represents a sink to which Zeebe will submitted records representing completed workflow state.  
```

Comment:
Yes, but not only completed workflow instances. Every record from the log can be exported.

Reply:
Noted.